components/renku_data_services/migrations/versions/8413f10ef77f_modify_stored_procedures.py

```diff
@@ -0,0 +1,79 @@
+"""modify stored procedures
+
+Revision ID: 8413f10ef77f
+Revises: fe3b7470d226
+Create Date: 2025-07-09 14:09:38.801974
+
+"""
+
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "8413f10ef77f"
+down_revision = "fe3b7470d226"
+branch_labels = None
+depends_on = None
+
+# NOTE: This combines the procedures introduced in migrations 89aa4573cfa9 and f4ad62b7b323.
+
+
+def upgrade() -> None:
+    op.execute("""CREATE OR REPLACE FUNCTION cleanup_after_slug_deletion()
+RETURNS TRIGGER AS
+$$
+BEGIN
+    IF OLD.project_id IS NOT NULL AND OLD.namespace_id IS NOT NULL AND OLD.data_connector_id IS NULL THEN
+        DELETE FROM projects.projects WHERE projects.id = OLD.project_id;
+    ELSIF old.project_id is NOT NULL AND old.namespace_id IS NOT NULL AND old.data_connector_id IS NOT NULL THEN
```

Comment:
This does not look correct: it won't delete DCs in user namespaces or group namespaces

```suggestion
    ELSIF old.namespace_id IS NOT NULL AND old.data_connector_id IS NOT NULL THEN
```

Reply:
Done in https://github.com/SwissDataScienceCenter/renku-data-services/pull/928/commits/3f7b17011402832795f0037222ed361cbff52741